src/teller/teller.go

```diff
@@ -25,16 +26,17 @@ type Teller struct {
 }
 
 // New creates a Teller
-func New(log logrus.FieldLogger, exchanger exchange.Exchanger, addrGen addrs.AddrGenerator, cfg config.Config) *Teller {
+func New(log logrus.FieldLogger, exchanger exchange.Exchanger, addrGen, ethAddrGen addrs.AddrGenerator, cfg config.Config) *Teller {
```

Comment:
Same comment about `AddrManager`, later we need to wrap these into one `AddrManager`, or require the caller to configure which `AddrManager` they want after they create.